src/classes/GiftBatch.cls

```diff
@@ -138,4 +139,15 @@ public inherited sharing class GiftBatch {
     public void updateRecords() {
         gifts.updateRecords();
     }
+
+    public Money total() {
+        //todo: use giftBatch Totals (and place in Selector layer)
+        String model = BGE_DataImportBatchEntry_CTRL.getDataImportModel(batchId);
+        BGE_DataImportBatchEntry_CTRL.DataImportModel typedModel =
+                (BGE_DataImportBatchEntry_CTRL.DataImportModel)
+                        JSON.deserialize(model,
+                BGE_DataImportBatchEntry_CTRL.DataImportModel.class);
+        return new Money(typedModel.totalRowAmount, 'US');
+    }
```

Comment:
This isn't pretty, but it is a way to begin to install the ideal model while still using the same underlying source code while putting characterization tests in place.  Once the characterization tests are in place, then this method can be safely refactored to simplify and/or use another mechanism to retrieve the total, like GIftBatchTotals.